### PR TITLE
Fixes issue #5100 Debugger tooltip not always displayed

### PR DESCRIPTION
**Pull request by mkrueger**
We shouldn't have to implement that service - it's not using anything special. So I request to move it.

https://github.com/dotnet/roslyn/issues/28041

For now copying the code 1:1 works.

### CODE REVIEW

File: main/src/addins/CSharpBinding/MonoDevelop.CSharp.Completion/CSharpCompletionTextEditorExtension.cs [hunk 1]
```diff
@@ -757,7 +757,18 @@ public Task<MonoDevelop.Ide.CodeCompletion.ParameterHintingResult> InternalHandl
 
 		async Task<DebugDataTipInfo> IDebuggerExpressionResolver.ResolveExpressionAsync (IReadonlyTextDocument editor, DocumentContext doc, int offset, CancellationToken cancellationToken)
 		{
-			return await Resolver.DebuggerExpressionResolver.ResolveAsync (editor, doc, offset, cancellationToken).ConfigureAwait (false);
+			var analysisDocument = doc.AnalysisDocument;
+			if (analysisDocument == null)
+				return default (DebugDataTipInfo);
+			var debugInfoService = doc.RoslynWorkspace.Services.GetLanguageServices (LanguageNames.CSharp).GetService<Microsoft.CodeAnalysis.Editor.Implementation.Debugging.ILanguageDebugInfoService> ();
```
Comment (Therzok) [inline main/src/addins/CSharpBinding/MonoDevelop.CSharp.Completion/CSharpCompletionTextEditorExtension.cs]: We could use this: http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Shared/Extensions/DocumentExtensions.cs,23

**Comment by slluis** [comment]
@mrward can you verify the fix?

**Comment by mrward** [comment]
Partially fixed. Hovering over exception variables inside the catch statement all work as expected.

If you hover over the catch line near the right hand side of the 'ex' you see the local variable tooltip instead of the debugger tooltip. If you hover over the left hand side you see the debugger tooltip. In VS Mac 7.5 the debugger tooltip is shown unless you hover over the ')' at the end of the catch. Not tried VS 2017 here.

debuggertooltip2

**Comment by mrward** [comment]
VS 2017's behaviour seems to the same as VS Mac 7.5.

**Comment by slluis** [comment]
@mkrueger did you change something? What's the status of this?

**Comment by mkrueger** [comment]
Should work IMO

**Comment by slluis** [comment]
@monojenkins backport release-7.6

**Comment by slluis** [comment]
@mkrueger can you backport to 7.6? thanks.

**Pull request merged**